File: Ridy-taxi-app/ios/driver/driver/Networking/HTTP/LocationUpdate.swift
//
//  LocationUpdate.swift
//  driver
//
//  Created by Manly Man on 11/25/19.
//  Copyright © 2019 minimal. All rights reserved.
//

import Foundation

import MapKit

public class LocationUpdate: HTTPRequest {
    public var path: String = "driver/update_location"
    
    public typealias ResponseType = EmptyClass
    public var params: [String : Any]?
    
    init(jwtToken: String, location: CLLocationCoordinate2D, inTravel: Bool = false) {
        self.params = [
            "token": jwtToken,
            "location": try! location.asDictionary(),
            "inTravel": inTravel
        ]
    }
}

File: Ridy-taxi-app/android/driver/src/main/java/com/innomalist/taxi/driver/networking/http/LocationUpdate.kt
package com.innomalist.taxi.driver.networking.http

import com.google.android.gms.maps.model.LatLng
import com.innomalist.taxi.common.networking.http.interfaces.HTTPRequest

class LocationUpdate(jwtToken: String, location: LatLng, inTravel: Boolean): HTTPRequest() {
    override val path: String = "driver/update_location"
    init {
        this.params = mapOf(
                "token" to jwtToken,
                "location" to mapOf("x" to location.longitude, "y" to location.latitude),
                "inTravel" to inTravel
        )
    }
}